### PR TITLE
Simplify image handling

### PR DESCRIPTION
**Pull request by apacheli**
Add a private utility method `_formatImage` to `Client` to simplify the process of getting a image.

### CODE REVIEW

File: lib/Client.js [hunk 1]
```diff
@@ -2064,6 +2064,21 @@ class Client extends EventEmitter {
         }));
     }
 
+    _formatImage(url, format, size) {
+        if(!format || !Constants.ImageFormats.includes(format.toLowerCase())) {
+            format = url.includes("a_") ? "gif": this.options.defaultImageFormat;
```
Comment (abalabahaha) [inline lib/Client.js]: `"/a_"` is a better string to check for. Still hacky, but eh...

**Review by abalabahaha — CHANGES_REQUESTED** [review]

File: lib/Client.js [hunk 2]
```diff
@@ -2064,6 +2064,21 @@ class Client extends EventEmitter {
         }));
     }
 
+    _formatImage(url, format, size) {
+        if(!format || !Constants.ImageFormats.includes(format.toLowerCase())) {
+            format = url.includes("/a_") ? "gif": this.options.defaultImageFormat;
+        }
+        if(
+            !size ||
```
Comment (abalabahaha) [inline lib/Client.js]: Bit of an odd if condition style

Comment (Apacheopteryx) [reply lib/Client.js]: It's this way to remain consistent to what it was before, the only difference having `!size ||`

Comment (apacheli) [reply lib/Client.js]: It's this way to remain consistent to what it was before, other than having `!size ||` in it

**Review by abalabahaha — DISMISSED** [review]

**Review by abalabahaha — APPROVED** [review]

File: lib/structures/GroupChannel.js [hunk 3]
```diff
@@ -72,7 +71,7 @@ class GroupChannel extends PrivateChannel { // (╯°□°）╯︵ ┻━┻
     }
 
     get iconURL() {
-        return this.icon ? `${Endpoints.CDN_URL}/channel-icons/${this.id}/${this.icon}.${this._client.options.defaultImageFormat}?size=${this._client.options.defaultImageSize}` : null;
+        return this.icon ? this._client._formatImage(Endpoints.GROUP_ICON(this.id, this.icon)) : null;
```
Comment (abalabahaha) [inline lib/structures/GroupChannel.js]: Ahem

File: lib/structures/GroupChannel.js [hunk 4]
```diff
@@ -81,17 +80,7 @@ class GroupChannel extends PrivateChannel { // (╯°□°）╯︵ ┻━┻
...
-        }
-        if(
-            size < ImageSizeBoundaries.MINIMUM ||
-            size > ImageSizeBoundaries.MAXIMUM ||
-            (size & (size - 1))
-        ) {
-            size = this._client.options.defaultImageSize;
-        }
-        return this.icon ? `${Endpoints.CDN_URL}/channel-icons/${this.id}/${this.icon}.${format}?size=${size}` : null;
+        return this.icon ? this._client._formatImage(Endpoints.GROUP_ICON(this.id, this.icon), format, size) : null;
```
Comment (abalabahaha) [inline lib/structures/GroupChannel.js]: hem

**Review by abalabahaha — CHANGES_REQUESTED** [review]

File: lib/structures/GroupChannel.js (hunk 4 above)
Comment (apacheli) [reply lib/structures/GroupChannel.js]: LOL

**Review by abalabahaha — APPROVED** [review]

**Pull request merged**